### PR TITLE
chore: Normalize implicit backslahes in single quoted strings internally

### PR DESCRIPTION
**Pull request by mvorisek**
related with https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7669 fixer introduce

### CODE REVIEW

File: tests/Fixer/Casing/ClassReferenceNameCasingFixerTest.php
Comment (mvorisek) [inline tests/Fixer/Casing/ClassReferenceNameCasingFixerTest.php:200]: Here you can see why consistent CS is important, here the CS was not consistent even on a single line!

**Comment by mvorisek** [comment]
Can someone please review and merge this PR? 🙏

**Comment by Wirone** [comment]
@mvorisek I won't do this, @keradus must decide if he wants this kind of change.

**Comment by mvorisek** [comment]
Hi @keradus, can you please give a green light to this PR?

**Comment by mvorisek** [comment]
@keradus can we get your approval here, please?

**Comment by mvorisek** [comment]
@Wirone this PR is here for 2 weeks already, can I ask you for an approval? It is not a new rule, double quoted strings are already normalized, this PR improves the PhpCsFixer config to normalize also single quoted strings as introduced in #7669.

**Comment by Wirone** [comment]
@mvorisek *no*, we have an agreement with @keradus that decisions like this require his review. I said it explicitly before and I don't understand why you ping me 🤷‍♂️. Well, I kind of understand, I also don't like when my changes wait long, but [look at my open PRs](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pulls?q=is%3Aopen+is%3Apr+author%3AWirone+sort%3Acreated-asc+-is%3Adraft) - the oldest one will be 1 year old in a few weeks time 🤷‍♂️. It does not work like this, everyone works at their own pace - nobody pings you for changes in the PRs when changes were requested, we just wait.

@keradus currently is not available and you just need to wait patiently, pinging won't help you, trust me 😉.

**Comment by mvorisek** [comment]
Hello maintainers, can we get your approval on this PR? Thank you in advance!

File: src/RuleSet/Sets/PhpCsFixerSet.php
Comment (keradus) [inline src/RuleSet/Sets/PhpCsFixerSet.php:122]: I see change of config as something unexpected, I would need to ask wider question in first place. That is not directly scope of this PR, but needed for me to align how we want to move on here.

why `string_implicit_backslashes` is having so many options? why for each syntax we allow to escape/ignore/unescape?

Why not just always escape for double_quoted and heredoc, always unescape for single_quoted (and maybe, maybe, allow to set them to `ignore`) ?

Out of the blue, I would rather deprecate all options of the rule, looking at it.

Comment (mvorisek) [reply src/RuleSet/Sets/PhpCsFixerSet.php:122]: Configs options are for users that want different syntax or possibly needs to align the syntax with another CS fixing tools.

In this PR, let's focus solely in normalizing single quotes. All other string notations are fixed already.

Comment (keradus) [reply src/RuleSet/Sets/PhpCsFixerSet.php:122]: I want to normalize only for direction that we consider the best practice.
do we have the best practice aligned? is it the current defaults of those options?

Comment (mvorisek) [reply src/RuleSet/Sets/PhpCsFixerSet.php:122]: Yes - this PR is to unquote possibly quoted single string backslashes and this is strongly the used style in this project and IMO other projects too.

**Comment by mvorisek** [comment]
@Wirone do you consider this PR approved by the discussion above?

**Comment by mvorisek** [comment]
Guys, can this PR be approved, please?

**Comment by mvorisek** [comment]
Can this PR be approved please?

**Review by keradus — COMMENTED** [review]
about https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7786#issuecomment-2041117628

related to https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7786#discussion_r1529086043


I tend to not give approval when I'm on -0.5 side of it, yet don't want to explicitly break it because maybe someone else want to give approval.

For me, ideal flow should be following:
- PR 1: change config _only_ in `.php-cs-fixer.dist.php`, can be merged with anyone to approve. I am on -0 for it, but not blocking it.
- PR 2: change config in RuleSet. We should align on each option of the rule, align what we consider a best practice for the PHP ecosystem (maybe mimic PER-CS or Symfony, maybe come to it by conclusion of Fixer's core team membersd
- PR 3: create TODOs to adjust defaults of rule itself

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thanks @mvorisek 🍻!

**Pull request merged**
